pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -881,8 +908,9 @@ func (os *OpenStack) getVolume(diskName string) (volumes.Volume, error) {
 	return volume, err
 }
 
+// getComputeIDbyHostname tries to find ID of the OpenStack instance where
+// this code runs.
 func (os *OpenStack) getComputeIDbyHostname(cClient *gophercloud.ServiceClient) (string, error) {
```

Comment:
I completely agree that the hostname matching ugly-ness should be removed. `instance-id` is already discovered by the kubelet during initial node [registration](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L872). I wanted to pass this `instance-id` from cinder volume plugin directly to the detach function.  Though `volume-host` is cloud aware it is not does not know the nodename. 

I did explore the config drive way, but I wasn't sure if it is ok to assume that it will be present everywhere. It would definitely address the original need. We could look at both, config drive and metadata server, if that works.